src/ServiceControl/MessageFailures/Handlers/UnArchiveMessagesHandler.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace ServiceControl.MessageFailures.Handlers
+{
+    using System.Linq;
+    using Contracts.MessageFailures;
+    using InternalMessages;
+    using NServiceBus;
+    using Raven.Client;
+
+    public class UnArchiveMessagesHandler : IHandleMessages<UnArchiveMessages>
+    {
+        public IDocumentSession Session { get; set; }
```

Comment:
Should this use doc store and manage it's own session? As we aren't executing in a transaction it's possible for the bulk update to fail if one of the messages gets updated after it's loaded and the publish will go out before the save fails. 

It's not terribly important at this stage because nothing should be happening to errors in the archived state. It may cause problems with really bad luck and retention deletion.